crates/astria-conductor/src/executor.rs

```diff
@@ -210,16 +228,25 @@ impl<C: ExecutionClient> Executor<C> {
             return Ok(None);
         }
 
-        if let Some(execution_hash) = self.sequencer_hash_to_execution_hash.get(&block.block_hash) {
+        if let Some(execution_block) = self
+            .sequencer_hash_to_execution_block
+            .get(&block.block_hash)
+        {
             debug!(
                 height = block.header.height.value(),
-                execution_hash = hex::encode(execution_hash),
+                execution_hash = hex::encode(&execution_block.hash),
                 "block already executed"
             );
-            return Ok(Some(execution_hash.clone()));
+            return Ok(Some(execution_block.clone()));
         }
 
-        let prev_block_hash = self.execution_state.clone();
+        let prev_block_hash = if let Some(soft_commitment) = self.commitment_state.soft.clone() {
+            soft_commitment.hash
+        } else {
+            warn!("could not get previous block. soft commitment is None");
+            return Ok(None);
```

Comment:
is there ever a case this could happen? seems concerning if there is, as nothing can get executed

Reply:
I don't think there is a case it would ever occur. That type comes from the proto definitions. I think I misunderstood your comment and added this complex handling instead of just using `wrap_err` haha. Should I just use `wrap_err` here?

Reply:
maybe return an error here - looking at the code, it seems this case shouldn't ever occur, so would be developer error if it does

Reply:
It should not ever happen no. What's a better way I can handle the `Option` here?

Reply:
It looks like executor is handling too many of the protobuf idiosyncracies like `soft` and `hard` being `Option`. If it basically never makes sense for these fields to be unset, then it's better to have one extra step that converts from the `protobuf-CommitmentState` to an `idiotmatic-CommitmentState` (failing if one of the fields is not set).

This way you don't have to worry about all of that `None` business.